moped-editor/src/views/projects/newProjectView/NewProjectMap.js

```diff
@@ -203,6 +247,81 @@ const NewProjectMap = ({
     saveActionDispatch
   );
 
+  /**
+   * Handles when a new map feature is selected and updates feature collection state
+   * @param {Object} selectedFeature - The selected feature geojson
+   * @return {undefined} Side effect of calling setFeatureCollection with new state
+   */
+  const handleSelectedFeatureUpdate = selectedFeature => {
+    const isPresent = isFeaturePresent(
+      selectedFeature,
+      featureCollection.features,
+      selectedFeature.properties.sourceLayer
+    );
+
+    if (selectedFeature.properties.sourceLayer !== "CTN" || isPresent) {
+      // we are only concerned with CTN (aka line) features that are being added to the feature collection
+      // otherwise handle normally
+      commitFeatureCollectionUpdate(
+        selectedFeature,
+        featureCollection,
+        setFeatureCollection,
+        isPresent
+      );
+      return;
+    }
+
+    // we must decide if the line the feature may be fragmented from tiling
+    const map = mapRef.current.getMap();
+    // get the current map bounds (LngLatBounds instance) and convert it to a simple array
+    const bbox = map
+      .getBounds()
+      .toArray()
+      .flat();
+    // convert the bbox array to a geojson line
+    const bboxLine = polygonToLine(bboxPolygon(bbox));
+    const intersectsWithBounds = booleanIntersects(bboxLine, selectedFeature);
+
+    if (!intersectsWithBounds) {
+      // feature is entirely contained—we have it's complete geometry
+      commitFeatureCollectionUpdate(
+        selectedFeature,
+        featureCollection,
+        setFeatureCollection,
+        isPresent
+      );
+      return;
+    }
+    // this feature is rendered to the edge of the viewport and may be fragmented. we cannot
+    // know with certainty, so we fetch it's complete geometry from AGOL to be safe.
+    // also considered turf.booleanWithin() and turf.booleanContains - not reliable.
+    const selectedFeatureId = getFeatureId(
+      selectedFeature,
+      selectedFeature.properties.sourceLayer
+    );
+
+    queryCtnFeatureService(selectedFeatureId).then(queriedFeatureCollection => {
```

Comment:
here's the AGOL API call—and the performance hit while fetch occurs. in a nutshell, we only query AGOL if 1) the selected feature is a CTN line and 2) it is touching the edge of the viewport and 3) the feature is being added (not removed) from the feature collection state.